stempy/image.cpp

```diff
@@ -578,6 +592,81 @@ RadialSum<uint64_t> radialSum(InputIt first, InputIt last, int scanWidth, int sc
   return radialSum;
 }
 
+namespace {
+// helper function that returns the binding result
+std::vector<float> getContainer(const STEMImage& inImage, const int numBins)
+{
+  // find the min and max value across all the input STEM images
+  auto min = std::numeric_limits<float>::max();
+  auto max = std::numeric_limits<float>::min();
+
+  // information about input STEMImage
+  int width = inImage.width;
+  int height = inImage.height;
+  auto curData = inImage.data;
+  std::cout << "Input STEM Image has width = " << width
+            << ", height = " << height << std::endl;
+
+  // curImage.data is shared_ptr<uint64_t []>
+  for (int i = 0; i < width * height; i++) {
+    // std::cout << "curData[" << i << "] = " << curData[i] << std::endl;
+    if (curData[i] < min)
+      min = curData[i];
+    if (curData[i] > max)
+      max = curData[i];
+  }
+
+  // the "length" of each slot of the container
+  float length = (float)((max - min) / numBins);
```

Comment:
Also, we usually try to use C++-style casts (`static_cast<float>()`) rather than C-style casts.

Reply:
Thank you for pointing it out, I have fixed it.